### PR TITLE
Port functions from old repositiory

### PR DESCRIPTION
**Pull request by ReubenHill**
Direct port of my implementation LibFTD2XX. Makes use of the library path provided by `deps.jl`

This adds various methods have been which make `FT_HANDLE` act more like a subtype of `IO`. This allows methods like `read` to work as expected without having to be explicitly defined.

To do:
 - [x] update `README.md`

### CODE REVIEW

**Comment by ReubenHill** [comment]
Since this fails checks, should I resolve #5 before merging?

**Comment by samuelpowell** [comment]
No it's okay, I meant to enable AppVeyor not Travis, I'll fix that

File: src/LibFTD2XX.jl [hunk 1]
```diff
@@ -10,4 +10,263 @@ else
...
+  FT_EEPROM_ERASE_FAILED,
+  FT_EEPROM_NOT_PRESENT,
+  FT_EEPROM_NOT_PROGRAMMED,
+  FT_INVALID_ARGS,
+  FT_NOT_SUPPORTED,
+  FT_OTHER_ERROR,
+  FT_DEVICE_LIST_NOT_READY,
+)
+
+include("rarelyused.jl")
```
Comment (samuelpowell) [inline src/LibFTD2XX.jl]: Does any of this stuff need to come with this PR?

File: src/LibFTD2XX.jl [hunk 2]
```diff
@@ -10,4 +10,263 @@ else
...
+const cfunc = Dict{Symbol, Ptr{Void}}()
+
+function __init__()
+  lib[] = Libdl.dlopen(libFTD2XX)
+  for n in cfuncn
+    cfunc[n] = Libdl.dlsym(lib[], n)
+  end
+end
+
+const DWORD = Cuint
```
Comment (samuelpowell) [inline src/LibFTD2XX.jl]: Move const above __init__

File: src/LibFTD2XX.jl [hunk 3]
```diff
@@ -10,4 +10,263 @@ else
...
+
+function destroy!(handle::FT_HANDLE)
+  if handle.p != C_NULL
+    flush(handle)
+    close(handle)
+  end
+  handle.p = C_NULL
+end
+
+@enum(
```
Comment (samuelpowell) [inline src/LibFTD2XX.jl]: move enum above functions

File: src/LibFTD2XX.jl [hunk 4]
```diff
@@ -10,4 +10,263 @@ else
...
+include("rarelyused.jl")
+
+@enum(
+  FTOpenBy,
+  OPEN_BY_SERIAL_NUMBER = FT_OPEN_BY_SERIAL_NUMBER,
+  OPEN_BY_DESCRIPTION = FT_OPEN_BY_DESCRIPTION,
+  OPEN_BY_LOCATION = FT_OPEN_BY_LOCATION,
+)
+
+const FT_PURGE_RX = 1
```
Comment (samuelpowell) [inline src/LibFTD2XX.jl]: Move to top also, and struct/const definitions

**Pull request merged**